### ISSUE TITLE
Unexpected default ordering when findAll() on Postgres

### ISSUE THREAD

**Issue by leodutra** [sequelize/sequelize#9289]
## What are you doing?
```js
<Model>.findAll()
```

## What do you expect to happen?
Retrieve list ordered by id ASC

## What is actually happening?
Not ordering output

__Dialect:__ postgres
__Dialect version:__ 7.4.1
__Database version:__ 9.6
__Sequelize version:__ 4.33.4
__Tested with latest release:__ No

**Comment by knoxcard** [comment]
I see, you'd think it be setup the "same way" for all, I agree that it should.

**Comment by leodutra** [comment]
Yes.
And I'm not sure if even findAll() without include is sorting.
I'll check as soon as possible.

**Comment by heisian** [comment]
I'm not sure if the SQL spec even guarantees an ordering. It's always best to be explicit about any ordering that you need.

**Comment by heisian** [comment]
See here:

https://stackoverflow.com/questions/6585574/postgres-default-sort-by-id-worldship

> Rows are returned in an unspecified order, per sql specs, unless you add an order by clause. In Postgres, that means you'll get rows in, basically, the order that live rows read on the disk.

AKA you DON'T get a guaranteed ordering unless explicitly specified.

**Comment by heisian** [comment]
@mickhansen If you'd be willing to provide Issues ACLs, I'd be happy to help bat down some support issues like so there's less noise among bug reports.

**Comment by mickhansen** [comment]
@heisian i'm not a decision maker anymore, /cc @sushantdhiman

**Comment by leodutra** [comment]
@heisian Man, thanks a lot for that stackoverflow answer. This explains why Postgres had a totally different behavior.

@mickhansen @sushantdhiman 
IMHO, documentation should have a tips section or common practices. This ordering question is something that would be pointed out there. The other things that should be pointed are:

- enforce `{ foreignKey: { allowNull: false } }` for relations where null is not expected (but the NO ACTION on delete);
- `underscore: true` expectations https://github.com/sequelize/sequelize/issues/6423;
- sequelize migrations does not create automigrations like Django, but there's a [lib](https://github.com/flexxnn/sequelize-auto-migrations) covering it (which Sequelize should embrace IMHO);
- async await [incompatibilities](https://github.com/othiym23/node-continuation-local-storage/issues/98) due to node-continuation-local-storage. Use cls-hooked for transactions on ES6+ node setups;
- way to safe query tables with sensitive information like password https://github.com/sequelize/sequelize/issues/1462. I solved this with `toJSON` and sequelize hooks... but it would be to have seen something like this in the docs.

Just went from zero to 5 services using Sequelize, in 2 organizations, and those were the most persistent issues/tricks across them. A lot of Google queries and Github issues (not really issues but questions)... again, IMHO.

I think this is one of the most important Node.js projects I've ever heard about and I totally trust in the great job you have been doing.

**Comment by heisian** [comment]
I agree that the Sequelize documentation leaves a lot to be desired. If I feel motivated, I could see them getting closer to these docs: https://laravel.com/docs/5.6/eloquent-relationships

They're incredibly concise, informative, and easy on the eyes all at the same time.
